src/Magento/FunctionalTestingFramework/Test/Objects/ActionObject.php

```diff
@@ -504,6 +504,9 @@ private function findAndReplaceReferences($objectHandler, $inputString)
                     throw new TestReferenceException("Could not resolve entity reference " . $inputString);
                 }
             }
+            elseif (is_array($replacement)) {
+                $replacement = '["' . implode('","', array_map('addSlashes',$replacement)) . '"]';
```

Comment:
@nathanjosiah my bad, I don't think I was very clear about this portion. I meant that your addition should be entirely inside the `elseif` on line 496; that `elseif` is the portion that deals with the object passed in being an `EntityDataObject`. Since the only time `$replacement` could only ever be an `array` is if it's an array inside an `EntityDataObject`, we should only every have to check it inside that conditional.

Does that make sense?

Reply:
Ah, yes I see what you mean. I have pushed an updated.